JoodHajeer/assignment2/JoodHajeer-2.cpp

```diff
@@ -89,21 +85,21 @@ BTNode<T> * Binary_Tree<T> ::find(T &VAL, BTNode<T> * node) {//return a pointer
 	return rightSide;
 }
 template <class T>
-bool Binary_Tree<T> :: insert( T &VAL,  T & parent_, char child_) {//insert node to a non-empty tree
-	//insert only unique values , child_ tells left or right child
+bool Binary_Tree<T> ::insert(T &VAL, T & PARENT, bool childDirection ) {//insert node to a non-empty tree
+	//insert only unique values , childDirection tells left =false  or right child =true
 	if (find(VAL))//value already exist in the tree, do not insert it
 		return false;
-	BTNode<T> * prevNode = find(parent_ , root);
+	BTNode<T> * prevNode = find(PARENT , root);
 	if (prevNode == nullptr)//parent node was not found, or tree is empty
 		return false; 
-	if (prevNode->left != nullptr && child_ == 'L') //parent already has a left child
+	if (prevNode->left != nullptr && childDirection == false) //parent already has a left child
 		return false;
-	if (prevNode->right != nullptr && child_ == 'R')//parent already has a right child
+	if (prevNode->right != nullptr && childDirection == true)//parent already has a right child
 		return false;
 	BTNode<T> * newNode = new BTNode<T>(VAL, nullptr, nullptr, prevNode , prevNode->depth+1); //val, left, right, patent, new depth
-	if (child_=='L') 
+	if (childDirection ==false)

```

Comment:
NIT: Comparing bools to true / false is kind of strange. Can use `if (!childDirection)` /  `if (childDirection)`
Same above.
For that to make more sense tough, the name of the bool should be indicative of what true/false means, so maybe better named as insertToTheRight, or similar.
(Ideally is better to use an enum, but don't worry :) )